tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
         self.assertAllClose(result, expected)
 
-    def test_dist_pairwise(self):
-
-        point_a = 1.0 / gs.sqrt(129.0) * gs.array([10.0, -2.0, -5.0, 0.0, 0.0])
-        point_b = 1.0 / gs.sqrt(435.0) * gs.array([1.0, -20.0, -5.0, 0.0, 3.0])
-
-        point = gs.array([point_a, point_b])
-        result = self.metric.dist_pairwise(point)
-
-        expected = gs.array([[0.0, 1.24864502], [1.24864502, 0.0]])
-
-        self.assertAllClose(result, expected, rtol=1e-3)
-
-    def test_dist_pairwise_parallel(self):
-        n_samples = 15
-        points = self.space.random_uniform(n_samples)
-        result = self.metric.dist_pairwise(points, n_jobs=2, prefer="threads")
-        is_sym = Matrices.is_symmetric(result)
-        belongs = Matrices(n_samples, n_samples).belongs(result)
-        self.assertTrue(is_sym)
-        self.assertTrue(belongs)
-
-    def test_dist_orthogonal_points(self):
-        # Distance between two orthogonal points is pi / 2.
-        point_a = gs.array([10.0, -2.0, -0.5, 0.0, 0.0])
-        point_a = point_a / gs.linalg.norm(point_a)
-        point_b = gs.array([2.0, 10, 0.0, 0.0, 0.0])
-        point_b = point_b / gs.linalg.norm(point_b)
-        result = gs.dot(point_a, point_b)
-        expected = 0
-        self.assertAllClose(result, expected)
-
-        result = self.metric.dist(point_a, point_b)
-        expected = gs.pi / 2
-
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_dist_and_projection_to_tangent_space(self):
-        base_point = gs.array([16.0, -2.0, -2.5, 84.0, 3.0])
-        base_point = base_point / gs.linalg.norm(base_point)
-        vector = gs.array([9.0, 0.0, -1.0, -2.0, 1.0])
-        tangent_vec = self.space.to_tangent(vector=vector, base_point=base_point)
-
-        exp = self.metric.exp(tangent_vec=tangent_vec, base_point=base_point)
-        result = self.metric.dist(base_point, exp)
-        expected = gs.linalg.norm(tangent_vec) % (2 * gs.pi)
+    def test_random_von_mises_fisher_belongs(self, dim, n_samples):
+        space = self.space(dim)
+        result = space.belongs(space.random_von_mises_fisher(n_samples=n_samples))
+        self.assertAllClose(gs.all(result), gs.array(True))
+
+    def test_random_von_mises_fisher_mean(self, dim, kappa, n_samples, expected, atol):
+        space = self.space(dim)
+        points = space.random_von_mises_fisher(kappa=kappa, n_samples=n_samples)
+        sum_points = gs.sum(points, axis=0)
+        result = sum_points / gs.linalg.norm(sum_points)
+        self.assertAllClose(result, expected, atol=atol)
+
+    def test_tangent_spherical_to_extrinsic(
+        self, dim, tangent_vec_spherical, base_point_spherical, expected
+    ):
+        space = self.space(dim)
+        result = space.tangent_spherical_to_extrinsic(
+            tangent_vec_spherical, base_point_spherical
+        )
         self.assertAllClose(result, expected)
 
-    def test_exp_and_dist_and_projection_to_tangent_space_vec(self):
```

Comment:
missing? this is the same as the above, but vectorized. Do we need it?

Reply:
already part of improvs